app/lib/core/view_models/home_screen_view_model.dart

```diff
@@ -19,6 +23,45 @@ class HomeScreenViewModel extends BaseViewModel {
       locator<AuthenticationService>();
   final NavigationService _navigationService = locator<NavigationService>();
 
+  StreamSubscription<Optional<User>> _currentUserSubscription;
+  StreamSubscription<Transporter> _transporterSubscription;
+  Transporter _transporter;
+
+  Transporter get transporter => _transporter;
+
+  @mustCallSuper
+  void dispose() {
+    _cancelTransporterSubscription();
+    if (_currentUserSubscription != null) _currentUserSubscription.cancel();
+    super.dispose();
+  }
+
+  void _cancelTransporterSubscription() {
+    if (_transporterSubscription != null) _transporterSubscription.cancel();
+  }
+
+  void loadTransporterInfo() {
+    final thisUser = _authenticationService.currentUser;
+    if (thisUser.isPresent()) {
+      _transporterSubscription = _databaseService
+          .getUpdatedTransporter(thisUser.get().uid)
+          .listen((account) {
+        _transporter = account;
+        notifyListeners();
+      });
+      _currentUserSubscription = _authenticationService.currentUserChanges
+          .listen((Optional<User> user) {
+        // User has logged out or is no longer authenticated, lock the ViewModel
+        if (!user.isPresent()) _cancelTransporterSubscription();
+      });
+    } else {
+      _dialogService.showDialog(
+        title: 'Launching the account screen failed',
```

Comment:
- [ ] Oops! Looks like this message was reused. Maybe call it `Launching the home screen failed`?

Reply:
Yea, forgot to change it. Thank you!